android/src/main/java/com/reactnativecommunity/webview/extension/file/Base64FileDownloader.kt

```diff
@@ -0,0 +1,169 @@
+package com.reactnativecommunity.webview.extension.file
+
+import android.Manifest
+import android.app.AlertDialog
+import android.content.ContentValues
+import android.content.Context
+import android.content.DialogInterface
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Environment
+import android.provider.MediaStore
+import android.util.Base64
+import android.util.Log
+import android.webkit.MimeTypeMap
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat
+import java.io.File
+import java.io.FileOutputStream
+
+internal object Base64FileDownloader {
+
+	/**
+	 * This method checks for WRITE_EXTERNAL_STORAGE permission for Android 23 - 29 and requests if necessary
+	 */
+	fun downloadBase64File(context: Context, base64: String, downloadingMessage: String, requestFilePermission: (String) -> Unit) {
+		val base64Parts = base64.split(",")
+		if (base64Parts.size != 2) {
+			Log.e("Base64FileDownloader", "Unable to parse base64 to download a file. Base64 value was: $base64")
+			return
+		}
+		val (mimeType, extension) = getMimeTypeAndFileExtension(base64Parts[0])
+
+		showAlertDialog(context, extension) {
+			Toast.makeText(context, downloadingMessage, Toast.LENGTH_LONG).show()
+			val fileBytes = Base64.decode(base64Parts[1], Base64.DEFAULT)
+			saveFileToDownloadsFolder(context, base64, fileBytes, mimeType, extension, requestFilePermission)
+		}
+	}
+
+	/**
+	 * This method is called from PermissionListener after WRITE_EXTERNAL_STORAGE permission is granted
+	 */
+	fun downloadBase64FileWithoutPermissionCheckAndDialog(
```

Comment:
Any thoughts on adding unit testing to these public functions. If the context is not too hard to mock it may be nice to add a bit of coverage.